public/js/home.js

```diff
@@ -52,6 +52,54 @@ window.addEventListener("click", function(event) {
         tableData_closeAllSelect();
     }
 });
+// detect color scheme and set slider
+window.addEventListener("load", loadMode);
+function loadMode() {
+    // loads current os mode and sets slider accordingly
+    const slider = document.getElementById("dark-check");
+    let currentMode = "";
+    if (window.matchMedia("(prefers-color-scheme: dark)").matches) {
+        document.body.classList.toggle("dark");
+        currentMode = "dark";
+        // sets the slider to match the actual mode
+        slider.checked = true;
+    }
+    else {
+        currentMode = "light";
+    }
+    if (localStorage.getItem("color-scheme") === null) {
+        var preferredMode = currentMode;
+    }
+    else {
+        var preferredMode = localStorage.getItem('color-scheme');
+    }
+    // sets mode to preferred if it's different than the current one
+    if (preferredMode !== currentMode) {
+        slider.click();
+        currentMode = preferredMode;
+    }
+    return currentMode;
+};
+
+// Toggle dark mode
+let dark = true;
+function darkMode() {
+    let r = document.querySelector(':root');
+    let rs = getComputedStyle(r);
+    let currentMode = document.querySelector("body").classList.contains("dark");
+    // If the color variables are set to dark mode, toggle light mode
+    if (currentMode) {
+        document.body.classList.toggle("dark");
+        localStorage.setItem("color-scheme", "light");
+        dark = false;
+    }
+    // Otherwise, toggle back to dark mode
+    else {
+        document.body.classList.toggle("dark");
+        localStorage.setItem("color-scheme", "dark");
+        dark = true;
+    }
+ }
```

Comment:
Simplify code by removing unused variables and needlessly verbose comments
```suggestion
// Toggle between dark and light mode
function darkMode() {
    document.body.classList.toggle("dark");
    localStorage.setItem("color-scheme",
        document.querySelector("body").classList.contains("dark") ?
            'dark' : 'light');
}
```